builder/azure/arm/step_deploy_template.go

```diff
@@ -279,82 +324,171 @@ func (s *StepDeployTemplate) deleteImage(ctx context.Context, imageName string,
 	if len(xs) < 3 {
 		return errors.New("Unable to parse path of image " + imageName)
 	}
-	_, err = s.client.GiovanniBlobClient.Delete(pollingContext, storageAccountName, blobName, giovanniBlobStorageSDK.DeleteInput{})
+	_, err = s.client.GiovanniBlobClient.Delete(pollingContext, "images", blobName, giovanniBlobStorageSDK.DeleteInput{})
 	return err
 }
 
-func (s *StepDeployTemplate) deleteDeploymentResources(ctx context.Context, subscriptionId, deploymentName, resourceGroupName string) error {
+func (s *StepDeployTemplate) deleteVirtualMachine(ctx context.Context, vmID virtualmachines.VirtualMachineId) error {
```

Comment:
While this requires a lot more code, I think its better to explicitly break apart these steps, the current function is really hard to debug when stuff goes wrong

Reply:
Most of the code looks similar, I'd think we can factorise it a bit more in order to reduce the amount of duplicated code, but that can be done later, no need to make this PR more complex today